In [1]:
import pandas as pd

In [2]:
data=pd.read_csv(r"C:\Users\win10\Desktop\Healthcare_dataset.csv")

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3424 entries, 0 to 3423
Data columns (total 69 columns):
 #   Column                                                              Non-Null Count  Dtype 
---  ------                                                              --------------  ----- 
 0   Ptid                                                                3424 non-null   object
 1   Persistency_Flag                                                    3424 non-null   object
 2   Gender                                                              3424 non-null   object
 3   Race                                                                3424 non-null   object
 4   Ethnicity                                                           3424 non-null   object
 5   Region                                                              3424 non-null   object
 6   Age_Bucket                                                          3424 non-null   object
 7   Ntm_Speciality          

In [5]:
data.shape

(3424, 69)

In [6]:
data.head()

Ptid Persistency_Flag  Gender           Race     Ethnicity   Region  \
0   P1       Persistent    Male      Caucasian  Not Hispanic     West   
1   P2   Non-Persistent    Male          Asian  Not Hispanic     West   
2   P3   Non-Persistent  Female  Other/Unknown      Hispanic  Midwest   
3   P4   Non-Persistent  Female      Caucasian  Not Hispanic  Midwest   
4   P5   Non-Persistent  Female      Caucasian  Not Hispanic  Midwest   

  Age_Bucket        Ntm_Speciality Ntm_Specialist_Flag  \
0        >75  GENERAL PRACTITIONER              Others   
1      55-65  GENERAL PRACTITIONER              Others   
2      65-75  GENERAL PRACTITIONER              Others   
3        >75  GENERAL PRACTITIONER              Others   
4        >75  GENERAL PRACTITIONER              Others   

       Ntm_Speciality_Bucket  ... Risk_Family_History_Of_Osteoporosis  \
0  OB/GYN/Others/PCP/Unknown  ...                                   N   
1  OB/GYN/Others/PCP/Unknown  ...                                   N   
2  OB/GYN/Others/PCP/Unknown  ...                                   N   
3  OB/GYN/Others/PCP/Unknown  ...                                   N   
4  OB/GYN/Others/PCP/Unknown  ...                                   N   

  Risk_Low_Calcium_Intake  Risk_Vitamin_D_Insufficiency  \
0                       N                             N   
1                       N                             N   
2                       Y                             N   
3                       N                             N   
4                       N                             N   

  Risk_Poor_Health_Frailty Risk_Excessive_Thinness  \
0                        N                       N   
1                        N                       N   
2                        N                       N   
3                        N                       N   
4                        N                       N   

  Risk_Hysterectomy_Oophorectomy Risk_Estrogen_Deficiency Risk_Immobilization  \
0                              N                        N                   N   
1                              N                        N                   N   
2                              N                        N                   N   
3                              N                        N                   N   
4                              N                        N                   N   

  Risk_Recurring_Falls Count_Of_Risks  
0                    N              0  
1                    N              0  
2                    N              2  
3                    N              1  
4                    N              1  

[5 rows x 69 columns]

In [7]:
data.dtypes

Ptid                              object
Persistency_Flag                  object
Gender                            object
Race                              object
Ethnicity                         object
                                   ...  
Risk_Hysterectomy_Oophorectomy    object
Risk_Estrogen_Deficiency          object
Risk_Immobilization               object
Risk_Recurring_Falls              object
Count_Of_Risks                     int64
Length: 69, dtype: object

CHECK FOR NULL VALUES

In [9]:
data.isnull().any().any()

False

CHECK FOR DUPLICATES

In [15]:
data.duplicated().any()

False

CHECK FOR VALUE COUNTS

In [16]:
data['Persistency_Flag'].value_counts()

Non-Persistent    2135
Persistent        1289
Name: Persistency_Flag, dtype: int64

In [26]:
data['Age_Bucket'].value_counts()

>75      1439
65-75    1086
55-65     733
<55       166
Name: Age_Bucket, dtype: int64

In [18]:
data['Race'].value_counts()

Caucasian           3148
Other/Unknown         97
African American      95
Asian                 84
Name: Race, dtype: int64

In [27]:
data['Region'].value_counts()

Midwest          1383
South            1247
West              502
Northeast         232
Other/Unknown      60
Name: Region, dtype: int64

VALUE COUNTS FOR DEMOGRAPHICS

In [28]:
data['Ethnicity'].value_counts()

Not Hispanic    3235
Hispanic          98
Unknown           91
Name: Ethnicity, dtype: int64

Maximum patients are Not Hispanic

In [29]:
data['Gender'].value_counts()

Female    3230
Male       194
Name: Gender, dtype: int64

In [30]:
data['Idn_Indicator'].value_counts()

Y    2557
N     867
Name: Idn_Indicator, dtype: int64

Provider Attributes

In [32]:
data['Ntm_Speciality_Bucket'].value_counts()

OB/GYN/Others/PCP/Unknown    2104
Endo/Onc/Uro                  716
Rheum                         604
Name: Ntm_Speciality_Bucket, dtype: int64

Maximum physician are OB/GYN

In [33]:
data['Tscore_Bucket_Prior_Ntm'].value_counts()

>-2.5     1951
<=-2.5    1473
Name: Tscore_Bucket_Prior_Ntm, dtype: int64

T Score of the patient at the time of the NTM Rx (within 2 years prior from rxdate) more in  2.5

In [34]:
data['Change_T_Score'].value_counts()

No change    1660
Unknown      1497
Worsened      173
Improved       94
Name: Change_T_Score, dtype: int64

In [36]:
data['Risk_Segment_During_Rx'].value_counts()

Unknown    1497
HR_VHR      965
VLR_LR      962
Name: Risk_Segment_During_Rx, dtype: int64

In [37]:
data['Change_Risk_Segment'].value_counts()

Unknown      2229
No change    1052
Worsened      121
Improved       22
Name: Change_Risk_Segment, dtype: int64

In [38]:
data['Count_Of_Risks'].value_counts()

1    1242
0     970
2     781
3     317
4      91
5      15
6       6
7       2
Name: Count_Of_Risks, dtype: int64

Clinical Factors

In [39]:
data['Dexa_Freq_During_Rx'].value_counts()

0      2488
5       114
6       107
7        93
8        71
4        68
10       55
12       52
3        46
14       38
9        32
11       30
1        24
2        24
13       19
20       15
16       14
18       14
22       13
26       10
24       10
15        9
30        7
17        7
28        7
21        7
36        5
19        3
42        3
32        3
34        3
52        2
48        2
58        2
25        2
39        2
88        2
54        1
146       1
50        1
35        1
44        1
108       1
72        1
40        1
68        1
45        1
38        1
69        1
118       1
66        1
110       1
33        1
23        1
27        1
81        1
37        1
29        1
Name: Dexa_Freq_During_Rx, dtype: int64

In [40]:
data['Dexa_During_Rx'].value_counts()

N    2488
Y     936
Name: Dexa_During_Rx, dtype: int64

In [41]:
data['Frag_Frac_Prior_Ntm'].value_counts()

N    2872
Y     552
Name: Frag_Frac_Prior_Ntm, dtype: int64

In [42]:
data['Frag_Frac_During_Rx'].value_counts()

N    3007
Y     417
Name: Frag_Frac_During_Rx, dtype: int64

In [43]:
data['Gluco_Record_During_Rx'].value_counts()

N    2522
Y     902
Name: Gluco_Record_During_Rx, dtype: int64

In [44]:
data['Gluco_Record_Prior_Ntm'].value_counts()

N    2619
Y     805
Name: Gluco_Record_Prior_Ntm, dtype: int64

In [45]:
data['Injectable_Experience_During_Rx'].value_counts()

Y    3056
N     368
Name: Injectable_Experience_During_Rx, dtype: int64

any injectable drug usage in the recent 12 months before the NTM OP Rx is 3056

Risk factors

In [59]:
data['Risk_Type_1_Insulin_Dependent_Diabetes'].value_counts()

N    3285
Y     139
Name: Risk_Type_1_Insulin_Dependent_Diabetes, dtype: int64

In [47]:
data['Risk_Osteogenesis_Imperfecta'].value_counts()

N    3421
Y       3
Name: Risk_Osteogenesis_Imperfecta, dtype: int64

In [48]:
data['Risk_Rheumatoid_Arthritis'].value_counts()

N    3294
Y     130
Name: Risk_Rheumatoid_Arthritis, dtype: int64

In [49]:
data['Risk_Untreated_Chronic_Hyperthyroidism'].value_counts()

N    3422
Y       2
Name: Risk_Untreated_Chronic_Hyperthyroidism, dtype: int64

In [50]:
data['Risk_Untreated_Chronic_Hypogonadism'].value_counts()

N    3297
Y     127
Name: Risk_Untreated_Chronic_Hypogonadism, dtype: int64

In [51]:
data['Risk_Untreated_Early_Menopause'].value_counts()

N    3412
Y      12
Name: Risk_Untreated_Early_Menopause, dtype: int64

In [52]:
data['Risk_Patient_Parent_Fractured_Their_Hip'].value_counts()

N    3168
Y     256
Name: Risk_Patient_Parent_Fractured_Their_Hip, dtype: int64

In [53]:
data['Risk_Smoking_Tobacco'].value_counts()

N    2780
Y     644
Name: Risk_Smoking_Tobacco, dtype: int64

In [54]:
data['Risk_Chronic_Malnutrition_Or_Malabsorption'].value_counts()

N    2954
Y     470
Name: Risk_Chronic_Malnutrition_Or_Malabsorption, dtype: int64

In [55]:
data['Risk_Chronic_Liver_Disease'].value_counts()

N    3406
Y      18
Name: Risk_Chronic_Liver_Disease, dtype: int64

In [56]:
data['Risk_Family_History_Of_Osteoporosis'].value_counts()

N    3066
Y     358
Name: Risk_Family_History_Of_Osteoporosis, dtype: int64

In [57]:
data['Risk_Low_Calcium_Intake'].value_counts()

N    3382
Y      42
Name: Risk_Low_Calcium_Intake, dtype: int64

In [60]:
data['Risk_Vitamin_D_Insufficiency'].value_counts()

N    1788
Y    1636
Name: Risk_Vitamin_D_Insufficiency, dtype: int64

In [61]:
data['Risk_Poor_Health_Frailty'].value_counts()

N    3232
Y     192
Name: Risk_Poor_Health_Frailty, dtype: int64

In [62]:
data['Risk_Excessive_Thinness'].value_counts()

N    3357
Y      67
Name: Risk_Excessive_Thinness, dtype: int64

In [63]:
data['Risk_Hysterectomy_Oophorectomy'].value_counts()

N    3370
Y      54
Name: Risk_Hysterectomy_Oophorectomy, dtype: int64

In [64]:
data['Risk_Estrogen_Deficiency'].value_counts()

N    3413
Y      11
Name: Risk_Estrogen_Deficiency, dtype: int64

In [65]:
data['Risk_Immobilization'].value_counts()

N    3410
Y      14
Name: Risk_Immobilization, dtype: int64

In [66]:
data['Risk_Recurring_Falls'].value_counts()

N    3355
Y      69
Name: Risk_Recurring_Falls, dtype: int64

# all risk factors have high N value than Y

OTHER FACTORS

In [68]:
data[['Comorb_Encounter_For_Screening_For_Malignant_Neoplasms','Comorb_Encounter_For_Immunization','Comorb_Vitamin_D_Deficiency','Comorb_Other_Joint_Disorder_Not_Elsewhere_Classified','Comorb_Encntr_For_Oth_Sp_Exam_W_O_Complaint_Suspected_Or_Reprtd_Dx','Comorb_Long_Term_Current_Drug_Therapy','Comorb_Dorsalgia','Comorb_Personal_History_Of_Other_Diseases_And_Conditions','Comorb_Other_Disorders_Of_Bone_Density_And_Structure','Comorb_Disorders_of_lipoprotein_metabolism_and_other_lipidemias','Comorb_Osteoporosis_without_current_pathological_fracture','Comorb_Personal_history_of_malignant_neoplasm','Comorb_Gastro_esophageal_reflux_disease','Concom_Cholesterol_And_Triglyceride_Regulating_Preparations','Concom_Narcotics','Concom_Systemic_Corticosteroids_Plain','Concom_Anti_Depressants_And_Mood_Stabilisers','Concom_Fluoroquinolones','Concom_Cephalosporins','Concom_Macrolides_And_Similar_Types','Concom_Broad_Spectrum_Penicillins','Concom_Anaesthetics_General','Concom_Viral_Vaccines']].value_counts()

Comorb_Encounter_For_Screening_For_Malignant_Neoplasms  Comorb_Encounter_For_Immunization  Comorb_Vitamin_D_Deficiency  Comorb_Other_Joint_Disorder_Not_Elsewhere_Classified  Comorb_Encntr_For_Oth_Sp_Exam_W_O_Complaint_Suspected_Or_Reprtd_Dx  Comorb_Long_Term_Current_Drug_Therapy  Comorb_Dorsalgia  Comorb_Personal_History_Of_Other_Diseases_And_Conditions  Comorb_Other_Disorders_Of_Bone_Density_And_Structure  Comorb_Disorders_of_lipoprotein_metabolism_and_other_lipidemias  Comorb_Osteoporosis_without_current_pathological_fracture  Comorb_Personal_history_of_malignant_neoplasm  Comorb_Gastro_esophageal_reflux_disease  Concom_Cholesterol_And_Triglyceride_Regulating_Preparations  Concom_Narcotics  Concom_Systemic_Corticosteroids_Plain  Concom_Anti_Depressants_And_Mood_Stabilisers  Concom_Fluoroquinolones  Concom_Cephalosporins  Concom_Macrolides_And_Similar_Types  Concom_Broad_Spectrum_Penicillins  Concom_Anaesthetics_General  Concom_Viral_Vaccines
N                                         

as we see N> Y